### PR TITLE
Cloud Build assets & system tests migration (AIP-47)

### PR DESCRIPTION
**Pull request by bkossakowska**
Cloud Build assets & system tests migration (AIP-47)

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/google/cloud/operators/cloud_build.py [hunk 1]
```diff
@@ -583,6 +650,10 @@ def execute(self, context: 'Context'):
             timeout=self.timeout,
             metadata=self.metadata,
         )
+
+        CloudBuildListLink.persist(
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply airflow/providers/google/cloud/operators/cloud_build.py:654]: This code was automatically reformatted by pre-commit

File: tests/system/providers/google/cloud/cloud_build/example_cloud_build.py [hunk 2]
```diff
@@ -37,94 +37,73 @@
...
-START_DATE = datetime(2021, 1, 1)
+ENV_ID = os.environ.get("SYSTEM_TESTS_ENV_ID")
+PROJECT_ID = os.environ.get("SYSTEM_TESTS_GCP_PROJECT")
 
-GCP_PROJECT_ID = os.environ.get("GCP_PROJECT_ID", "aitflow-test-project")
+DAG_ID = "example_gcp_cloud_build"
 
-GCP_SOURCE_ARCHIVE_URL = os.environ.get("GCP_CLOUD_BUILD_ARCHIVE_URL", "gs://airflow-test-bucket/file.tar.gz")
-GCP_SOURCE_REPOSITORY_NAME = os.environ.get("GCP_CLOUD_BUILD_REPOSITORY_NAME", "airflow-test-repo")
+BUCKET_NAME = f"bucket_{DAG_ID}-{ENV_ID}"
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply tests/system/providers/google/cloud/cloud_build/example_cloud_build.py]: I removed BUCKET_NAME and I changed the name of bucket.

File: tests/system/providers/google/cloud/cloud_build/example_cloud_build.py [hunk 3]
```diff
@@ -37,94 +37,73 @@
...
 create_build_from_storage_body = {
     "source": {"storage_source": GCP_SOURCE_ARCHIVE_URL},
-    "steps": [
-        {
-            "name": "gcr.io/cloud-builders/docker",
-            "args": ["build", "-t", f"gcr.io/$PROJECT_ID/{GCP_SOURCE_BUCKET_NAME}", "."],
-        }
-    ],
-    "images": [f"gcr.io/$PROJECT_ID/{GCP_SOURCE_BUCKET_NAME}"],
+    "steps": [{"name": "ubuntu", "args": ['echo', 'Hello world']}],
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply tests/system/providers/google/cloud/cloud_build/example_cloud_build.py:72]: This code was automatically reformatted by pre-commit

File: tests/system/providers/google/cloud/cloud_build/example_cloud_build.py [hunk 4]
```diff
@@ -37,94 +37,73 @@
...
 create_build_from_repo_body: Dict[str, Any] = {
-    "source": {"repo_source": {"repo_name": GCP_SOURCE_REPOSITORY_NAME, "branch_name": "main"}},
-    "steps": [
-        {
-            "name": "gcr.io/cloud-builders/docker",
-            "args": ["build", "-t", "gcr.io/$PROJECT_ID/$REPO_NAME", "."],
-        }
-    ],
-    "images": ["gcr.io/$PROJECT_ID/$REPO_NAME"],
+    "source": {"repo_source": {"repo_name": GCP_SOURCE_REPOSITORY_NAME, "branch_name": "master"}},
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply tests/system/providers/google/cloud/cloud_build/example_cloud_build.py:78]: This code was automatically reformatted by pre-commit

File: tests/system/providers/google/cloud/cloud_build/example_cloud_build.py [hunk 5]
```diff
@@ -37,94 +37,73 @@
...
+        task_id="upload_file",
+        src=FILE_LOCAL_PATH,
+        dst="file.tar.gz",
+        bucket=BUCKET_NAME_SRC,
+    )
 
     # [START howto_operator_create_build_from_storage]
     create_build_from_storage = CloudBuildCreateBuildOperator(
-        task_id="create_build_from_storage", project_id=GCP_PROJECT_ID, build=create_build_from_storage_body
+        task_id="create_build_from_storage", project_id=PROJECT_ID, build=create_build_from_storage_body
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply tests/system/providers/google/cloud/cloud_build/example_cloud_build.py:103]: This code was automatically reformatted by pre-commit.

File: tests/system/providers/google/cloud/cloud_build/example_cloud_build.py [hunk 6]
```diff
@@ -137,7 +116,7 @@
 
     # [START howto_operator_create_build_from_repo]
     create_build_from_repo = CloudBuildCreateBuildOperator(
-        task_id="create_build_from_repo", project_id=GCP_PROJECT_ID, build=create_build_from_repo_body
+        task_id="create_build_from_repo", project_id=PROJECT_ID, build=create_build_from_repo_body
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply tests/system/providers/google/cloud/cloud_build/example_cloud_build.py:116]: This code was automatically reformatted by pre-commit.

File: tests/system/providers/google/cloud/cloud_build/example_cloud_build.py [hunk 7]
```diff
@@ -150,14 +129,14 @@
 
     # [START howto_operator_list_builds]
     list_builds = CloudBuildListBuildsOperator(
-        task_id="list_builds", project_id=GCP_PROJECT_ID, location="global"
+        task_id="list_builds", project_id=PROJECT_ID, location="global"
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply tests/system/providers/google/cloud/cloud_build/example_cloud_build.py:129]: This code was automatically reformatted by pre-commit.

File: tests/system/providers/google/cloud/cloud_build/example_cloud_build.py [hunk 8]
```diff
@@ -167,101 +146,72 @@
...
-        trigger=update_build_trigger_body,
-    )
-    # [END howto_operator_create_build_trigger]
 
-    # [START howto_operator_get_build_trigger]
-    get_build_trigger = CloudBuildGetBuildTriggerOperator(
-        task_id="get_build_trigger",
-        project_id=GCP_PROJECT_ID,
-        trigger_id=create_build_trigger.output['id'],
+    delete_bucket_src = GCSDeleteBucketOperator(
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply tests/system/providers/google/cloud/cloud_build/example_cloud_build.py:175]: This code was automatically reformatted by pre-commit.

File: tests/system/providers/google/cloud/cloud_build/example_cloud_build.py [hunk 9]
```diff
@@ -167,101 +146,72 @@
...
+        task_id="delete_bucket_src", bucket_name=BUCKET_NAME_SRC, trigger_rule=TriggerRule.ALL_DONE
     )
-    # [END howto_operator_get_build_trigger]
 
-    # [START howto_operator_delete_build_trigger]
-    delete_build_trigger = CloudBuildDeleteBuildTriggerOperator(
-        task_id="delete_build_trigger",
-        project_id=GCP_PROJECT_ID,
-        trigger_id=create_build_trigger.output['id'],
+    (
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply tests/system/providers/google/cloud/cloud_build/example_cloud_build.py]: changed

File: tests/system/providers/google/cloud/cloud_build/example_cloud_build_trigger.py [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+    CloudBuildRunBuildTriggerOperator,
+    CloudBuildUpdateBuildTriggerOperator,
+)
+
+# START_DATE = datetime(2021, 1, 1)
+
+ENV_ID = os.environ.get("SYSTEM_TESTS_ENV_ID")
+PROJECT_ID = os.environ.get("SYSTEM_TESTS_GCP_PROJECT")
+
+DAG_ID = "0123_example_gcp_cloud_build_trigger"
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply tests/system/providers/google/cloud/cloud_build/example_cloud_build_trigger.py]: Done

File: tests/system/providers/google/cloud/cloud_build/example_cloud_build_trigger.py [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+
+GCP_SOURCE_REPOSITORY_NAME = os.environ.get("GCP_CLOUD_BUILD_REPOSITORY_NAME", "test-cloud-build-repo")
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply tests/system/providers/google/cloud/cloud_build/example_cloud_build_trigger.py]: changed

File: tests/system/providers/google/cloud/cloud_build/example_cloud_build_trigger.py [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+    "trigger_template": {
+        "project_id": PROJECT_ID,
+        "repo_name": GCP_SOURCE_REPOSITORY_NAME,
+        "branch_name": "main",
+    },
+    "filename": "example_cloud_build.yaml",
+}
+# [END howto_operator_gcp_create_build_trigger_body]
+
+update_build_trigger_body = {
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply tests/system/providers/google/cloud/cloud_build/example_cloud_build_trigger.py:60]: changed

File: tests/system/providers/google/cloud/cloud_build/example_cloud_build_trigger.py [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+    "trigger_template": {
+        "project_id": PROJECT_ID,
+        "repo_name": GCP_SOURCE_REPOSITORY_NAME,
+        "branch_name": "master",
+    },
+    "filename": "example_cloud_build.yaml",
+}
+
+# [START howto_operator_create_build_from_repo_body]
+create_build_from_repo_body: Dict[str, Any] = {
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply tests/system/providers/google/cloud/cloud_build/example_cloud_build_trigger.py:72]: This code was automatically reformatted by pre-commit

File: tests/system/providers/google/cloud/cloud_build/example_cloud_build_trigger.py [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+with models.DAG(
+    DAG_ID,
+    schedule_interval='@once',
+    start_date=datetime(2021, 1, 1),
+    catchup=False,
+    tags=["example"],
+) as dag:
+
+    # [START howto_operator_create_build_trigger]
+    create_build_trigger = CloudBuildCreateBuildTriggerOperator(
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply tests/system/providers/google/cloud/cloud_build/example_cloud_build_trigger.py:88]: This code was automatically reformatted by pre-commit

File: tests/system/providers/google/cloud/cloud_build/example_cloud_build_trigger.py [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+    # [START howto_operator_delete_build_trigger]
+    delete_build_trigger = CloudBuildDeleteBuildTriggerOperator(
+        task_id="delete_build_trigger",
+        project_id=PROJECT_ID,
+        trigger_id=create_build_trigger.output['id'],
+    )
+    # [END howto_operator_delete_build_trigger]
+
+    # [START howto_operator_list_build_triggers]
+    list_build_triggers = CloudBuildListBuildTriggersOperator(
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply tests/system/providers/google/cloud/cloud_build/example_cloud_build_trigger.py:128]: This code was automatically reformatted by pre-commit

File: tests/system/providers/google/cloud/cloud_build/example_cloud_build.py [hunk 16]
```diff
@@ -37,94 +37,73 @@
...
-GCP_PROJECT_ID = os.environ.get("GCP_PROJECT_ID", "aitflow-test-project")
+DAG_ID = "example_gcp_cloud_build"
 
-GCP_SOURCE_ARCHIVE_URL = os.environ.get("GCP_CLOUD_BUILD_ARCHIVE_URL", "gs://airflow-test-bucket/file.tar.gz")
-GCP_SOURCE_REPOSITORY_NAME = os.environ.get("GCP_CLOUD_BUILD_REPOSITORY_NAME", "airflow-test-repo")
+BUCKET_NAME = f"bucket_{DAG_ID}-{ENV_ID}"
+BUCKET_NAME_SRC = f"test-bucket_{DAG_ID}-{ENV_ID}"
+
+GCP_SOURCE_ARCHIVE_URL = os.environ.get("GCP_CLOUD_BUILD_ARCHIVE_URL", f"gs://{BUCKET_NAME_SRC}/file.tar.gz")
+GCP_SOURCE_REPOSITORY_NAME = os.environ.get(
```
[the remark this answers is not in the document]
Comment (bkossakowska) [reply tests/system/providers/google/cloud/cloud_build/example_cloud_build.py]: Done

File: airflow/providers/google/cloud/operators/cloud_build.py (hunk 1 above)
Comment (VladaZakharova) [inline airflow/providers/google/cloud/operators/cloud_build.py:654]: Can you please add indentation here for code readability?

**Review by lwyszomi — APPROVED** [review]

File: airflow/providers/google/cloud/operators/cloud_build.py (hunk 1 above)
Comment (bkossakowska) [reply airflow/providers/google/cloud/operators/cloud_build.py]: This code was automatically reformatted by pre-commit

**Pull request closed without merging**